satpy/readers/yaml_reader.py

```diff
@@ -888,18 +896,25 @@ def _load_dataset(dsid, ds_info, file_handlers, dim='y'):
         res.attrs = combined_info
         return res
 
-    def _load_area_def(self, dsid, file_handlers):
+    def _load_area_def(self, dsid, file_handlers, pad_data=True):
         """Load the area definition of *dsid* with padding."""
-        # Pad missing segments between the first available and expected
-        area_defs = _pad_later_segments_area(file_handlers, dsid)
+        if not pad_data:
+            return _load_area_def(dsid, file_handlers)
```

Comment:
Why not use `super` and make a extra function ?

Reply:
Because:
1. it didn't need to be a method
2. `@staticmethod` would've caused other problems
3. I couldn't get it to work with `super()`

In short, this was the easiest and fastest way to get it to work :grin: